lightningnetwork/lnd
# lnrpc: Add metadata field to `AddInvoice` and `DecodePayReq`

## Change Description
Add the ability to set the invoice `Metadata` field in the `AddInvoice` RPC and to see said metadata via `DecodePayReq`. This is very useful when needing to send some bytes with the invoice while still allowing the user to use the Memo field for messages

## Steps to Test
1. Create an invoice using the `AddInvoice` RPC and using the `metadata` field.
2. Decode said invoice with `DecodePayReq` and observe the `metadata` field.

## Pull Request Checklist
### Testing

### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
